src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -36,19 +38,107 @@ internal static bool IsEnabled()
             //from DiagnosticListener right after the check. So some requests happening right after subscription starts
             //might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instumented
 
-            Guid loggingRequestId = Guid.NewGuid();
-            LogHttpRequest(request,  loggingRequestId);
+            Task<HttpResponseMessage> responseTask;
 
-            Task<HttpResponseMessage> responseTask = base.SendAsync(request, cancellationToken);
-
-            LogHttpResponse(responseTask, loggingRequestId);
+            //We need to know if activity events are enabled and send Request/Response events otherwise
+            if (s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ActivityName))
+            {
+                if (Activity.Current != null)
+                {
+                    responseTask = SendInstrumentedAsync(request, cancellationToken);
+                }
+                else
+                {
+                    //null Activity.Current means that incoming request was not instrumented
+                    //and there is nothing we can do for outgoing request. 
+                    //Activity events are enabled so we don't send Request/Response events
```

Comment:
I want Activity events to substitute Request/Response events eventually. Activity events are fired at the same time and provide more flexible filtering and Activity.Current for context.
So yes, Activity events are mutually exclusive with Request/Response.
There will be listeners listening to Activity requests only and there are existing listeners for Request/Response events.
If consumer is subscribed for Activity events and Activity.Current is null, it means that current operation is not instrumented, so it turns off *any* diagnostics for this particular request.
If consumer wants to receive Request/Response events, they should subscribe for them only.

There may be a problem if there are 2 consumers in the same process, one listening to Activity and other listening to Request/Response. In this case, second one will never receive any events.